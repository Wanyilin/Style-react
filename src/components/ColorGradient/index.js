import React from 'react';
import { convertColorSet } from 'src/utils/utils';
import PaintPreview from 'src/components/PaintPreview';

import './index.scss';

const PositionSpot = ({ hexText, alpha, position }) => (
	<div className="color-position">
		<div className="circle" />
		<div className="position">{ position }</div>
		<div className="color-display">
			<span className="color-box" style={{ background: `${hexText}` }} />
			<span className="hex-text">{ hexText }</span>
			<span>{ alpha }</span>
		</div>
	</div>
);

const ColorGradient = ({
	name,
	value,
	type,
}) => {
	const colorsSet = convertColorSet(value);
	const paintValue = [...colorsSet];
	const firstColor = colorsSet.shift();
	return (
		<>
			<PaintPreview type={type} value={paintValue} name={name} />
			<div className="gradient">
				<PositionSpot
					hexText={firstColor.hexText}
					alpha={firstColor.alpha}
					position={firstColor.position}
				/>
				{colorsSet.map((colorSet) => (
					<div key={`${colorSet.hexText}-set`}>
						<div className="line" />
						<PositionSpot hexText={colorSet.hexText} alpha={colorSet.alpha} position={colorSet.position}/>
					</div>
				))}
			</div>
		</>
	)
}

export default ColorGradient;
