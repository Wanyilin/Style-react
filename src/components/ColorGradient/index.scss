@import 'src/theme.scss';

@mixin circle {
	width: 6px;
	height: 6px;
	border: 1px solid #fff;
	border-radius: 50%;
}


.gradient {
	margin-bottom: $ListItemMargin;
	padding-left: 3px;

	.color-position-spot {
		display: flex;
    color: #fff;
    align-items: center;
    height: 28px;

		
		.circle {
			@include circle;
			
			&::before {
				height: 19px;
				border-left: 1px solid rgba(255, 255, 255, 0.4);
				margin-left: 2.5px;
				content: '';
				display: block;
				position: relative;
				top: 8px;
			}
		}

		&:last-child {
			.circle::before {
				content: unset;
			}
		}

		.position {
			width: 32px;
			margin: 0 8px;
		}

		.color-display, .position {
			@include FontStyle($ItemContentSize, $ItemTitleWeight, $MainFontColor);
			letter-spacing: 0.01em;
		}
		
		.color-display {
			display: flex;
			align-items: center;
			margin-left: 8px;
		}

		.paint-preview {
			margin-bottom: 0;
			margin-right: 4px;
		}
		
	}
}