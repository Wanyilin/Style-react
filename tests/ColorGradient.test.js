import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ColorGradient from '../src/components/ColorGradient';


jest.mock('../src/utils/utils', () => ({
	convertColorSet: jest.fn((value) => value),
}))


describe('Test ColorGradient', () => {
	test('Should match snapshot', () => {
		const value = ['color1', 'color2'];
		const component = renderer.create(
			<ColorGradient type="type" value={value} name="name" />
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	test('Should display context as expect', () => {
		const value = ['color1', 'color2'];
		const { container } = render(
			<ColorGradient type="type" value={value} name="name" />
		);
		const lineNode = container.getElementsByClassName('line');
		const circleNode = container.getElementsByClassName('circle');
		expect(lineNode.length).toEqual(1);
		expect(circleNode.length).toEqual(2);
	})
 });
 
