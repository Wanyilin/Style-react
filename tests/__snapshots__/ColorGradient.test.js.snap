// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ColorGradient Should match snapshot 1`] = `
[
  <div
    className="paint-preview category-title"
  >
    <span
      className="color-box"
      style={
        {
          "background": "url(color1,color2)",
        }
      }
    />
    <span
      className="item-content"
    >
      color1
      color2
    </span>
  </div>,
  <div
    className="gradient"
  >
    <div
      className="color-position"
    >
      <div
        className="circle"
      />
      <div
        className="position"
      />
      <div
        className="color-display"
      >
        <span
          className="color-box"
          style={
            {
              "background": "undefined",
            }
          }
        />
        <span
          className="hex-text"
        />
        <span />
      </div>
    </div>
    <div>
      <div
        className="line"
      />
      <div
        className="color-position"
      >
        <div
          className="circle"
        />
        <div
          className="position"
        />
        <div
          className="color-display"
        >
          <span
            className="color-box"
            style={
              {
                "background": "undefined",
              }
            }
          />
          <span
            className="hex-text"
          />
          <span />
        </div>
      </div>
    </div>
  </div>,
]
`;
